### PR TITLE
fix(NewGroupConversation) - refactoring & focus on Confirmation page 

### LINKED ISSUE #7584: Final step of conversation creation doesn't focus
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. `+`
2. Enter a name
3. Tick "Allow guests"
4. Use keyboard tab to navigate to the next page
5. Tab through the user list and navigate to the next page

### Expected behaviour
The dialog is focused at the top

### Actual behaviour
Finish button is still focused (as it's the same button we just change the label) and pressing tab jumps to the message input in the background

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #7584

The focus for every page is the following 
- First Page: conversation Name text field
- Second Page: search input
- Third Page: Copy Link button

### 🚧 Tasks

- [ ] Code review

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

**Review by Antreesy — CHANGES_REQUESTED** [review]
Actually, `<Confirmation />` component is a pile of props and boilerplate, without complex logic.
If we get rid of it and move the code to the `<NewGroupConversation />`, it will be just 30 lines of template + 3 lines of script + 15 lines of styles.

And that could allow us to access buttons directly, and watch after `success` and `error` - if they are truthy, we'll focus an appropriate button

File: src/components/LeftSidebar/NewGroupConversation/Confirmation/Confirmation.vue [hunk 1]
```diff
@@ -87,6 +88,12 @@ export default {
 		},
 	},
 
+	mounted() {
+		const interval = setTimeout(() => {
+			this.$refs.copyLink.$el.focus()
+		}, 500);
+	},
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/LeftSidebar/NewGroupConversation/Confirmation/Confirmation.vue:95]: And don't forget to lint files, you have a semicolon

File: src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue [hunk 2]
```diff
@@ -94,14 +94,35 @@
...
 				<!-- Third page -->
-				<Confirmation v-else-if="page === 2"
-					class="new-group-conversation__content"
-					:token="newConversation.token"
-					:conversation-name="conversationNameTrimmed"
-					:error="error"
-					:is-loading="isLoading"
-					:success="success"
-					:is-public="isPublic" />
+				<div v-else-if="page === 2" class="confirmation new-group-conversation__content">
```
Comment (Antreesy) [inline src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue:97]:
```suggestion
				<div v-else-if="page === 2" class="new-group-conversation__content">
```

We could omit that class, as it doesn't have any styles applied

File: src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue [hunk 3]
```diff
@@ -265,6 +285,15 @@ export default {
 				this.passwordProtect = false
 			}
 		},
+
+		success(value) {
```
Comment (Antreesy) [inline src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue:289]: Let's also add the same, but watch for `error` and focus `Close` button instead, as it'll be only available

**Review by Antreesy — APPROVED** [review]
Tested on web, and works so far!
Let's add one more button in focus, and squash commits into one (or two)

**Comment by nickvergessen** [comment]
27 Backport is in #9954 
26 is canceled, so removing the label

**Pull request merged**